src/ForceHighPoly.Core/ForceHighPoly.Hooks.cs

```diff
@@ -1,36 +1,224 @@
 ﻿using HarmonyLib;
-using System;
 
 namespace KK_Plugins
 {
     public partial class ForceHighPoly
     {
         internal static class Hooks
         {
-            [HarmonyPrefix, HarmonyPatch(typeof(AssetBundleManager), nameof(AssetBundleManager.LoadAsset), typeof(string), typeof(string), typeof(Type), typeof(string))]
-            private static void LoadAssetPrefix(ref string assetName) => RemoveLow(ref assetName);
-
-            [HarmonyPrefix, HarmonyPatch(typeof(AssetBundleManager), nameof(AssetBundleManager.LoadAssetAsync), typeof(string), typeof(string), typeof(Type), typeof(string))]
-            private static void LoadAssetAsyncPrefix(ref string assetName) => RemoveLow(ref assetName);
-
             /// <summary>
-            /// Strip "_low" from the end of an asset name
+            /// Test all coordiantes parts to check if a low poly doesn't exist.
+            /// if low poly doesnt exist for an item set HiPoly to true and exit;
             /// </summary>
-            private static void RemoveLow(ref string assetName)
+            /// <param name="__instance"></param>
+            [HarmonyPostfix, HarmonyPatch(typeof(ChaControl), nameof(ChaControl.Initialize))]
```

Comment:
Does this fire when the character changes clothes on the main map? e.g. in dressing room. If not, then the old value would be used which might be wrong.

Reply:
Fired upon character creation, since game currently hates reloading the character